pay-api/src/pay_api/services/non_sufficient_funds.py

```diff
@@ -0,0 +1,192 @@
+# Copyright © 2019 Province of British Columbia
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Service to manage Non-Sufficient Funds."""
+from __future__ import annotations
+
+from typing import Optional
+
+from flask import current_app
+
+from pay_api.models import Invoice as InvoiceModel
+from pay_api.models import InvoiceReference as InvoiceReferenceModel
+from pay_api.models import InvoiceSchema, NonSufficientFundsModel, NonSufficientFundsSchema
+from pay_api.models import Payment as PaymentModel
+from pay_api.models import PaymentAccount as PaymentAccountModel
+from pay_api.models import PaymentSchema, db
+
+
+class NonSufficientFundsService:  # pylint: disable=too-many-instance-attributes,too-many-public-methods
+    """Service to manage Non-Sufficient Funds related operations."""
+
+    def __init__(self):
+        """Initialize the service."""
+        self.__dao = None
+        self._id: int = None
+        self._invoice_id: int = None
+        self._description: Optional[str] = None
+
+    @property
+    def _dao(self):
+        if not self.__dao:
+            self.__dao = NonSufficientFundsModel()
+        return self.__dao
+
+    @_dao.setter
+    def _dao(self, value):
```

Comment:
This pattern isn't ideal, alot of pylint ignores / boiler plate for very little gain

Yes it exists